FeedSX/screens/createPost/index.tsx

```diff
@@ -0,0 +1,549 @@
+import {
+  View,
+  Platform,
+  TouchableOpacity,
+  SafeAreaView,
+  Text,
+  ScrollView,
+} from 'react-native';
+import React, {useState} from 'react';
+import {launchImageLibrary} from 'react-native-image-picker';
+import DocumentPicker from 'react-native-document-picker';
+import {requestStoragePermission} from '../../utils';
+import {useAppSelector} from '../../store/store';
+import {
+  LMButton,
+  LMCarousel,
+  LMDocument,
+  LMHeader,
+  LMIcon,
+  LMImage,
+  LMInputText,
+  LMLinkPreview,
+  LMProfilePicture,
+  LMText,
+  LMVideo,
+} from '../../../LikeMinds-ReactNative-Feed-UI';
+import Layout from '../../constants/Layout';
+import {
+  DOCUMENT_ATTACHMENT_TYPE,
+  IMAGE_ATTACHMENT_TYPE,
+  LINK_ATTACHMENT_TYPE,
+  VIDEO_ATTACHMENT_TYPE,
+} from '../../constants/Strings';
+import {DecodeURLRequest} from 'likeminds-sdk';
+import {
+  getDecodedUrl,
+  setUploadAttachments,
+} from '../../store/actions/createPost';
+import {useDispatch} from 'react-redux';
+import {NavigationService} from '../../navigation';
+
+// selected image/video format interface
+interface imageDefaultFormat {
+  fileName: string;
+  fileSize: number;
+  height: number;
+  type: string;
+  uri: string;
+  width: number;
+  bitrate?: number;
+  duration?: number;
+  originalPath?: string;
+}
+// selected document format interface
+interface docDefaultFormat {
+  name: string;
+  size: number;
+  fileCopyUri: null;
+  type: string;
+  uri: string;
+}
```

Comment:
move this interface to different file and rename `docDefaultFormat` to `DocumentMetaData`

Reply:
resolved